backend/typescript/utilities/logger.ts

```diff
@@ -23,4 +23,31 @@ if (process.env.NODE_ENV !== "production") {
   );
 }
 
+export const logfactory = (filename: string): winston.Logger => {
```

Comment:
> Is there any concern that the factory would be inefficient?

Yeah about this, could we try something similar to this comment: https://github.com/winstonjs/winston/issues/197#issuecomment-10438446 instead. I'd rather not create a new logger for every file if possible

Let's rename the old logger in `logger.ts` to `_logger` and then create a new `logger` like this:
```ts
export const logger = (fileName: string) => {
  return {
    error: (message: string) => {
      _logger.error(`[$fileName}] {message}`);
    },
    info: (message: string) => {
      _logger.info(...);
    }
  }
}
```

Sorry for not bringing this up earlier but hopefully should be a quick change.

Reply:
Factory has been replaced by a function which supplies wrappers for all logger levels.